compiler/src/iree/compiler/InputConversion/MHLO/ConvertMHLOToLinalgExt.cpp

```diff
@@ -470,6 +471,140 @@ struct ReverseOpConversion : public OpConversionPattern<mhlo::ReverseOp> {
+                indexingMaps, iterators,
+                [=](OpBuilder &b, Location loc, ValueRange args) {
+                  // Indicies increase along the k dimension
+                  Value index = b.create<linalg::IndexOp>(loc, kDim);
+                  Value intIndex =
+                      b.create<arith::IndexCastOp>(loc, b.getI32Type(), index);
+                  b.create<linalg::YieldOp>(loc, intIndex);
+                })
+            .getResult(0);
+
+    // Create and initialize output tensors for LinalgExt TopK results
+    // Define the output types based on the results of CHLO TopK
+    ShapedType outputValuesType = op.getResultTypes()[0].dyn_cast<ShapedType>();
+    ShapedType outputIndicesType =
+        op.getResultTypes()[1].dyn_cast<ShapedType>();
+    dynSizes.clear();
+    for (auto en : llvm::enumerate(outputValuesType.getShape())) {
+      if (en.value() == ShapedType::kDynamicSize) {
+        dynSizes.push_back(rewriter.create<tensor::DimOp>(
+            loc, adaptor.getOperands()[0], en.index()));
+      }
+    }
+    Value initTensorOutputValues = rewriter.create<mlir::linalg::InitTensorOp>(
+        loc, dynSizes, outputValuesType.getShape(),
+        outputValuesType.getElementType());
+    Value initTensorOutputIndices = rewriter.create<mlir::linalg::InitTensorOp>(
+        loc, dynSizes, outputIndicesType.getShape(),
+        outputIndicesType.getElementType());
+    // Check that the last dim size in the output = k
+    if (outputValuesType.getShape().back() != k ||
+        outputIndicesType.getShape().back() != k) {
+      return failure();
+    }
+    // Initialize indices to 0 and values to negative infinity
+    Attribute negInfAttr;
+    if (outputValuesType.getElementType().isa<IntegerType>()) {
+      negInfAttr = rewriter.getIntegerAttr(
+          outputValuesType.getElementType(),
+          APInt::getSignedMinValue(outputValuesType.getElementType()
+                                       .cast<IntegerType>()
+                                       .getWidth()));
+    } else {
+      auto negApFloat = APFloat::getInf(outputValuesType.getElementType()
+                                            .cast<FloatType>()
+                                            .getFloatSemantics(),
+                                        true);
+      negInfAttr =
+          rewriter.getFloatAttr(outputValuesType.getElementType(), negApFloat);
+    }
+    Value negInf = rewriter.create<arith::ConstantOp>(
+        loc, negInfAttr, outputValuesType.getElementType());
+    Value zero = rewriter.create<arith::ConstantOp>(
+        loc, rewriter.getZeroAttr(outputIndicesType.getElementType()));
+    Value negInfTensor =
+        rewriter.create<linalg::FillOp>(loc, negInf, initTensorOutputValues)
+            .getResult(0);
+    Value zeroTensor =
+        rewriter.create<linalg::FillOp>(loc, zero, initTensorOutputIndices)
+            .getResult(0);
```

Comment:
Do we have to initialize the indices tensor? I think initializing output values is enough.

Reply:
Input indices I thought were the indices that are already set. These needn't be initialized. If these are in-out values they need to be passed to the `out` operands. Maybe I missed this on linalg ext op definition....

Reply:
Dont' think I fully follow the comment? Also https://github.com/google/iree/pull/9162#discussion_r879791272 might have more context? 

The input indices are initialized with index values. Thats separate from the output tensors which both need to be provided as inputs with initialized values. Are you saying the input values tensor which is set to 0 doesn't need the `linalg.fill` operation?